### PR TITLE
IssuanceDate and ExpirationDate

### PR DESCRIPTION
**Pull request by mkhraisha**
PR tackles #913 
Updated Diagrams, Explainer, V2 context csv and used the npm generate script, removed the index.html deprecation warnings.

***
<a href="https://pr-preview.s3.amazonaws.com/Mavennet/vc-data-model/pull/964.html" title="Last updated on Oct 27, 2022, 6:55 PM UTC (b1d5a42)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/964/55078af...Mavennet:b1d5a42.html" title="Last updated on Oct 27, 2022, 6:55 PM UTC (b1d5a42)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -4892,11 +4873,11 @@ <h3>Base Context</h3>
         "credentialStatus": {"@id": "cred:credentialStatus", "@type": "@id"},
         "credentialSubject": {"@id": "cred:credentialSubject", "@type": "@id"},
         "evidence": {"@id": "cred:evidence", "@type": "@id"},
-        "expirationDate": {"@id": "cred:expirationDate", "@type": "xsd:dateTime"},
+        "validUntil": {"@id": "cred:validUntil", "@type": "xsd:dateTime"},
         "holder": {"@id": "cred:holder", "@type": "@id"},
         "issued": {"@id": "cred:issued", "@type": "xsd:dateTime"},
         "issuer": {"@id": "cred:issuer", "@type": "@id"},
-        "issuanceDate": {"@id": "cred:issuanceDate", "@type": "xsd:dateTime"},
+        "validFrom": {"@id": "cred:validFrom", "@type": "xsd:dateTime"},
```
Comment (dlongley) [inline index.html:4880]: Note that `validUntil` and `validFrom` are already present in the context, at the end -- so these are duplicates.

Comment (mkhraisha) [reply index.html]: good catch, should be resolved now.

**Review by decentralgabe — APPROVED** [review]
was just about to do this 😄 

only question - do the existing context's need to be maintained for backwards compatibility?

**Comment by mkhraisha** [comment]
> was just about to do this 😄 
> 
> 
> 
> only question - do the existing context's need to be maintained for backwards compatibility?

I believe so, this PR only affects  v2.

File: vocab/credentials/v2/vocabulary.csv [hunk 2]
```diff
@@ -17,9 +17,9 @@ property,credentialSchema,,,credential schema,,cred:VerifiableCredential,cred:Cr
 property,credentialStatus,,,credential status,,cred:VerifiableCredential,cred:CredentialStatus,,The value of the `credentialStatus` property MUST be an instance of Credential Status.
 property,credentialSubject,,,credential subject,,cred:VerifiableCredential,,,An entity about which claims are made.
 property,evidence,,,evidence,,cred:VerifiableCredential,cred:CredentialEvidence,,The value of the `evidence` property MUST be one or more Credential Evidence instances.
-property,expirationDate,,,expiration date,,cred:VerifiableCredential,xsd:dateTime,,The value of the `expirationDate` property MUST be a string value of an ISO8601 combined date and time string representing the date and time the credential ceases to be valid.
```
Comment (msporny) [inline vocab/credentials/v2/vocabulary.csv:20]: Note that everything below here is going to conflict with https://github.com/w3c/vc-data-model/pull/968/files.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2022-11-02-vcwg#section1-1) on 2022-11-02

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.1. IssuanceDate and ExpirationDate (pr vc-data-model#964)

_See github pull request [vc-data-model#964](https://github.com/w3c/vc-data-model/pull/964)._




**Manu Sporny:** simultaneously ivan raised a PR to update vocabularies.  

_See github pull request [vc-data-model#968](https://github.com/w3c/vc-data-model/pull/968)._




**Manu Sporny:** and they conflict with one another.  
… Ivan and Mahmoud you have to talk to each other.  
… that is the only thing that keeps those PRs from being merged.  

**Ivan Herman:** my PR does two things. one yaml is now the language and the other thing it also includes the changes of property names.  
… should have made that more clear.  
… as far as i could understand it duplicates what Mahmoud has done.  
… apologies for that.  
… that is where we are.  

**Manu Sporny:** that is great because we have more people doing PRs.  

</details>

**Comment by iherman** [comment]
@mkhraisha I was looking at this in view of the conflicts with #968.

The conflicts are exclusively on the vocabulary part, ie, `vocab/credentials/v2/vocabulary.*` files (#968 does not touch those).

The problem is that #968 pretty much changes the way the ttl/jsonld/html files are generated. In the older version the starting point was the CSV file, whereas in the new version of the vocabulary generation the starting point is a YAML file. Also, I am not sure whether you changed the ttl/jsonld/html files by hand or you ran the generation scripts; if the former, that is a problem by itself, those files are generated. 

#968 also takes care of the core issue of your PR, namely the new properties are introduced. Technically, that PR does not _remove_ the old properties but, instead, _deprecates_ them; because the vocabulary is already deployed, it is not a good idea to simply remove them from the vocabulary, thereby creating dangling links on the linked data world.

To move forward: would it be possible for you to roll back all the changes you have made on  `vocab/credentials/v2/vocabulary.*`? (I am not sure technically what is the easiest way of doing so, @msporny, who is also a git-wizzard, may help in doing that.). If you are successful at that, then the conflicts disappear and the two PR-s can be merged independently of one another. 

If that is too complicated, then another approach is that this PR is merged, and I can then either rebase my changes on (the new) `main` branch, or I simply remove #968 and create a new PR with the necessary changes. 

Whichever is simpler for you (unless @msporny comes up with a better idea).

**Comment by mkhraisha** [comment]
@iherman probably just easiest to close this PR and if the issue still persists after your PR is merged then I can recreate it. This was not a lot of work for me.
If that works please close it or let me know and I'll close it.

**Comment by msporny** [comment]
> @iherman probably just easiest to close this PR and if the issue still persists after your PR is merged then I can recreate it. This was not a lot of work for me. If that works please close it or let me know and I'll close it.

Closing this PR in favor of #968. Once we merge #968, we can come back to this one.

**Pull request closed without merging**